test/fft_base.py

```diff
@@ -374,17 +374,20 @@ def _test_raise_excep_ifft(test_case,inarr,outarr,other_args=None):
             except KeyError:
                 delta = new_args.pop('delta_f')
                 new_args.update({'delta_t' : delta})
-        in_badkind = type(inarr)(pycbc.types.zeros(len(inarr)),dtype=_bad_dtype[dtype(outarr).type],
+        in_badkind = type(inarr)(pycbc.types.zeros(len(inarr)),
+                                 dtype=_bad_dtype[dtype(outarr).type],
                                  **new_args)
-        args = [in_badkind,outarr,tc.backends]
-        tc.assertRaises(ValueError,pycbc.fft.ifft,*args)
+        args = [in_badkind, outarr]
+        #pycbc.fft.ifft(in_badkind, outarr)
+        if str(outarr.dtype) not in ['complex64', 'complex128']:
+            tc.assertRaises((ValueError, KeyError), pycbc.fft.ifft, *args)
```

Comment:
@ahnitz Here's the first problem I flagged. Although I have currently put in an if statement here to make this test run.

Basically here we do an IFFT of a *real* input array to a *complex* output array (R2C IFFT). The tests previously expected this to raise a failure message (presumable R2C FFT was allowed but R2C IFFT is not). However, this appears to run happily now, so currently I am bypassing this test for R2C IFFTs. Is this the correct and expected behaviour? Are there cases where a C2R IFFT *is* sane for us ... maybe it would be used for determining the autocorrelation length of a PSD?